vignettes/fims-path-maturity.Rmd

```diff
@@ -0,0 +1,614 @@
+
+
+Within the rcpp_maturity.hpp file, we can link to the `fims::Information` class in order to register maturity parameters and specify whether or not they are random effects:
+
+```{Rcpp, class.source = "rcppchunk", eval = FALSE}
+//file: rcpp_maturity.hpp
+
+//d0 is a shared pointer that points to fims::Information
+std::shared_ptr<fims::Information<TMB_FIMS_REAL_TYPE> > d0 =
+        fims::Information<TMB_FIMS_REAL_TYPE>::GetInstance();
+ if (this->median.estimated) {
+      if (this->median.is_random_effect) {
+        /*
+         if median is estimated and a random effect, 
+         the median value from LogisticMaturityInterface (lm0->median) 
+         is passed to the Information member function RegisterRandomEffect
+        */
+        d0->RegisterRandomEffect(lm0->median);
+      } else {
+        /*
+         if median is estimated and not a random effect, 
+         the median value from LogisticMaturityInterface (lm0->median)
+         is passed to the Information member function RegisterParameter
+         */
+        d0->RegisterParameter(lm0->median);
+      }
+    }
+```
+
+
+Linking to the `fims::LogisticMaturity` class allows us to link the values input from R with the values used in model calculation:
+```{Rcpp, class.source = "rcppchunk", eval = FALSE}
+//file: rcpp_maturity.hpp
+
+//lm0 is a shared pointer that points to fims::LogisticMaturity
+std::shared_ptr<fims::LogisticMaturity<TMB_FIMS_REAL_TYPE> > lm0 =
+        std::make_shared<fims::LogisticMaturity<TMB_FIMS_REAL_TYPE> >();
+ 
+/*
+ the median value from LogisticMaturity (lm0->median)
+ equals the median value from LogisticMaturityInterface (this->median.value)
+ */
+lm0->median = this->median.value;
+```
+
+We can also link these two pointers together so that the **fims::Information** class links up with the **fims::LogisticMaturity**, but more details on this later.
+```{Rcpp, class.source = "rcppchunk", eval = FALSE}
+//file: rcpp_maturity.hpp
+
+/*
+ the maturity_models pointer from Information that matches the 
+ id of the fims::LogisticMaturity class is equal to the pointer 
+ to fims::LogisticMaturity 
+ */
+d0->maturity_models[lm0->id] = lm0;
+```
+
+
+The `add_to_fims_tmb()` function repeats code four times to track the estimated value of parameters along with their first, second, and third derivatives. 
```

Comment:
suggestion: can you format "add_to_fims_tmb()" to make it orange or whatever color the Rcpp interface is in the diagram? At first glance, I thought it was an R function exposed to the user, then I looked in the diagram and saw that it was orange. 😮‍💨 and felt kind of dumb.